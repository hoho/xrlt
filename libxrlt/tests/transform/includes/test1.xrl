<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">
    <xrl:response>
        <xrl:include>
            <xrl:href>/test/href/1</xrl:href>
            <xrl:type>xml</xrl:type>
            <xrl:with-header name="Content-Type">Some type</xrl:with-header>
            <xrl:with-header name="Nope" test="false()" select="100500" />
            <xrl:with-header name="Content-Length" test="true()" select="100500" />
            <xrl:with-param name="par1">p&amp;1</xrl:with-param>
            <xrl:with-param name="par2" select="'p;2'" />
            <xrl:with-param name="par3" test="false()" select="'p3'" />
            <xrl:with-param>
                <xrl:test select="true()" />
                <xrl:name>par4</xrl:name>
                <xrl:value select="'p&amp;4'" />
            </xrl:with-param>
            <xrl:success select="concat(/hello, /hello)" />
        </xrl:include>

        <xrl:include>
            <xrl:href>/test/href/2</xrl:href>
            <xrl:type select="'TEXT'" />
            <xrl:with-body>ololololo</xrl:with-body>
            <xrl:success select="concat(substring-before(/, 'many'), 'no',
                                        substring-after(text(), 'many'))" />
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/3'" />
            <xrl:method>POST</xrl:method>
            <xrl:type>json</xrl:type>
            <xrl:with-param body="yes" name="test">aa</xrl:with-param>
            <xrl:with-param body="yes" name="test2" select="'\bb/'" />
        </xrl:include>
    </xrl:response>
</xrl:requestsheet>
