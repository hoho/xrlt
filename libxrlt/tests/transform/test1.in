id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:0, error: 0, data:{"aaa
id:3, type:2, last:1, error: 0, data:bbb": 123, "ccc": true}
id:0, type:0, last:0, error: 0, data:
id:2, type:2, last:0, error: 0, data:<hello>
id:2, type:2, last:0, error: 0, data:world
id:2, type:2, last:1, error: 0, data:</hello>
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:there are
id:1, type:2, last:0, error: 0, data: many clouds
id:1, type:2, last:1, error: 0, data: in the sky
id:0, type:0, last:0, error: 0, data:
