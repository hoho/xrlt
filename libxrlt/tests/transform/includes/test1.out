XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: XML
sr header: Content-Type: Some type
sr header: Content-Length: 100500
sr url: /test/href/1
sr query: par1=p%261&par2=p%3B2&par4=p%264
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 2
sr method: GET
sr type: TEXT
sr url: /test/href/2
sr query: (null)
sr body: ololololo
XRLT_STATUS_SUBREQUEST
sr id: 3
sr method: POST
sr type: JSON
sr url: /test/href/3
sr query: (null)
sr body: test=aa&test2=%5Cbb%2F
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: worldworld
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: there are no clouds in the sky
chunk: 123
chunk: true
XRLT_STATUS_DONE
